aiida_optimize/_optimization_workchain.py

```diff
@@ -151,7 +152,10 @@ def finalize(self):  # pylint: disable=inconsistent-return-statements
         with self.optimizer() as opt:
             if not opt.is_finished_ok:
                 return self.exit_codes.ERROR_ENGINE_FAILED
-            optimal_process_output = opt.result_value
+            optimal_process_input = opt.result_input_value
+            optimal_process_input.store()
```

Comment:
Since the input was already passed to a sub-process, it should already be stored. Maybe we can replace this with `assert optimal_process_input.is_stored`?

Reply:
I think this was left over from before I implemented getting the input from the `_result_mapping`, in which case it was not already stored. I'll replace with the `assert`